main.go

```diff
@@ -248,12 +270,16 @@ func main() {
 	if *mappingConfig != "" {
 		err := mapper.InitFromFile(*mappingConfig, *cacheSize)
 		if err != nil {
-			panic(fmt.Sprintf("Error loading config: %s", err))
+			level.Error(logger).Log("msg", "error loading config", "error", err)
+			os.Exit(1)
 		}
 		if *dumpFSMPath != "" {
 			err := dumpFSM(mapper, *dumpFSMPath, logger)
 			if err != nil {
-				panic(fmt.Sprintf("Error dumping FSM: %s", err))
+				level.Error(logger).Log("msg", "error dumping FSM", "error", err)
+				// Failure to dump the FSM is an error (the user asked for it and it
+				// didn't happen) but not fatal (the exporter is fully functional
+				// afterwards).
```

Comment:
As an aside, should this always exit anyway? This sounds more like a tool than what a daemon should do in passing.

Reply:
hmm it could, yeah, although then I'd prefer to start a mapper CLI which could also do config validation.

Reply:
cf. #263